### PR TITLE
Support OP_RETURN data in Bitcoin transaction

### LINKED ISSUE #113: allow crafting btc tx with OP_RETURN
Following the delivery of https://ledgerhq.atlassian.net/browse/LIVE-4054 can we please expose this through the wallet-api? ty

### PR DESCRIPTION
**Pull request by JunichiSugiura**
close #113 

- `BitcoinTrasaction` type now supports `opReturnData` field
- Only hex string support while raw transaction type is `Buffer`
- Validation
  - buffer: 80 bytes
  - in ascii: 80 chars
  - in hex string: 160 chars

### CODE REVIEW

File: packages/core/src/families/bitcoin/serializer.ts [hunk 1]
```diff
@@ -1,30 +1,35 @@
...
 
 export function serializeBitcoinTransaction({
   family,
   amount,
   recipient,
   feePerByte,
+  opReturnData,
 }: BitcoinTransaction): RawBitcoinTransaction {
-  return {
+  return schemaRawBitcoinTransaction.parse({
```
Comment (Justkant) [inline packages/core/src/families/bitcoin/serializer.ts:12]: I don't think you should parse here as it's not done like that in other families and we do the validation on the server when receiving a `rawTransaction` in the `params`

**Review by Justkant — CHANGES_REQUESTED** [review]
Small thing about the parse in the serializer and you might want to recreate the changeset as it's not correctly detected by the bot

File: packages/core/src/families/bitcoin/serializer.ts (hunk 1 above)
Comment (JunichiSugiura) [reply packages/core/src/families/bitcoin/serializer.ts:12]: Found it. It makes sense.

https://github.com/LedgerHQ/wallet-api/blob/main/packages/server/src/internalHandlers/transaction.ts#L19

File: packages/core/src/families/bitcoin/validation.ts [hunk 2]
```diff
@@ -4,4 +4,5 @@ import { schemaFamilies, schemaTransactionCommon } from "../common";
 export const schemaRawBitcoinTransaction = schemaTransactionCommon.extend({
   family: z.literal(schemaFamilies.enum.bitcoin),
   feePerByte: z.string().optional(),
+  opReturnData: z.string().max(80).optional(),
```
Comment (IAmMorrow) [inline packages/core/src/families/bitcoin/validation.ts:7]: 80 bytes in a buffer != 80 characters in a utf8 string. Is this intended ?

We tend to transport raw data has hex strings. When it is the case, please include "hex" in the field's name so we know we are dealing with serialized data :)  We would also need to calculate the appropriate max string size for hex data.

File: packages/core/tests/serializers.spec.ts [hunk 3]
```diff
@@ -135,37 +135,69 @@ describe("serializers.ts", () => {
...
-      it("should succeed to serialize a bitcoin transaction without feePerByte", () => {
-        const transaction: BitcoinTransaction = {
-          amount: new BigNumber(100),
-          recipient: "recipient",
-          family,
-        };
-        const serializedTransaction = serializeTransaction(transaction);
+      describe("opReturnData", () => {
+        const data =
+          "We're no strangers to love You know the rules and so do I A full commitment's what I'm thinking of You wouldn't get this from any other guy...";
```
Comment (IAmMorrow) [inline packages/core/tests/serializers.spec.ts:173]: Hahahahaha good stuff

**Review by IAmMorrow — DISMISSED** [review]

File: packages/core/tests/serializers.spec.ts (hunk 3 above)
Comment (IAmMorrow) [reply packages/core/tests/serializers.spec.ts:173]: (this is not a request for change I just loled)

File: packages/core/src/families/bitcoin/validation.ts (hunk 2 above)
Comment (JunichiSugiura) [reply packages/core/src/families/bitcoin/validation.ts]: From your Slack response

```
1 hex character = 4 bits
2 hex characters = 8 bits = 1 byte
80 bytes = 160 hex characters
```

File: packages/simulator/tests/server.spec.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import { BitcoinTransaction, schemaFamilies } from "@ledgerhq/wallet-api-core";
+import BigNumber from "bignumber.js";
+import { getSimulatorTransport, profiles } from "../src";
+
+function createBitcoinTx(): BitcoinTransaction {
+  return {
+    family: schemaFamilies.enum.bitcoin,
+    amount: new BigNumber(0),
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    recipient: profiles.STANDARD.accounts[0]!.address,
```
Comment (Justkant) [inline packages/simulator/tests/server.spec.ts:11]: We can probably use `as const` in the profiles to avoid that kind of issue with typescript

**Review by Justkant — APPROVED** [review]
LGTM but we can probably try the `as const` in this PR

**Comment by JunichiSugiura** [comment]
@Justkant do you know what it means?
Screenshot 2023-03-07 at 15 23 04

File: packages/simulator/tests/server.spec.ts (hunk 4 above)
Comment (JunichiSugiura) [reply packages/simulator/tests/server.spec.ts]: I casted `as Account` instead.

Screenshot 2023-03-07 at 15 26 34

**Review by Justkant — APPROVED** [review]

**Pull request merged**